hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconContainerManager.java

```diff
@@ -203,6 +207,15 @@ private void checkContainerStateAndUpdate(ContainerID containerID,
         && isHealthy(state)) {
       LOG.info("Container {} has state OPEN, but given state is {}.",
           containerID, state);
+      Pipeline pipeline =
+              pipelineManager.getPipeline(containerInfo.getPipelineID());
+      // subtract open container count from the pipeline
+      int curCnt = pipelineToOpenContainer.getOrDefault(pipeline.getId(), 0);
+      if (curCnt == 1) {
+        pipelineToOpenContainer.remove(pipeline.getId());
+      } else if (curCnt > 0) {
+        pipelineToOpenContainer.put(pipeline.getId(), curCnt - 1);
+      }
```

Comment:
Can we use `containerInfo.getPipelineID()` directly?
Seems we are converting from `PipelineID` to `Pipeline` back to the same `PipelineID`?